File: study/lua/src/app/scene/view/guild/GuildMyRedPacketItemCell.lua

local ListViewCellBase = require("app.ui.ListViewCellBase")
local GuildMyRedPacketItemCell = class("GuildMyRedPacketItemCell", ListViewCellBase)
local TypeConvertHelper = require("app.utils.TypeConvertHelper")
local DataConst =  require("app.const.DataConst")
local GuildConst =  require("app.const.GuildConst")

function GuildMyRedPacketItemCell:ctor()
    self._textRedPacketName = nil
    self._resInfo = nil
	local resource = {
		file = Path.getCSB("GuildMyRedPacketItemCell", "guild"),
		binding = {
			_imageRedPacket = {
				events = {{event = "touch", method = "_onButton"}}
			},
		}
	}
	GuildMyRedPacketItemCell.super.ctor(self, resource)
end

function GuildMyRedPacketItemCell:onCreate()
	local size = self._resourceNode:getContentSize()
	self:setContentSize(size.width, size.height)

    self._imageRedPacket:setSwallowTouches(false)
end

function GuildMyRedPacketItemCell:update(data)
    local config = data:getConfig()
    local money = data:getTotal_money() *  data:getMultiple()
    local state = data:getRed_bag_state()

    self._textRedPacketName:setString(config.name)
   
    self._resInfo:updateUI(TypeConvertHelper.TYPE_RESOURCE,DataConst.RES_DIAMOND,money)
    self._resInfo:setTextColor(Colors.BRIGHT_BG_TWO)
    self._resInfo:showResName(false)

    if state == GuildConst.GUILD_RED_PACKET_NO_SEND then
       -- self._imageBg:loadTexture(Path.getCommonImage("img_com_board04b"))
        self:updateImageView("_imageBg", { visible = true, texture = Path.getCommonImage("img_com_board04b") })
        self._textStageName:setString(Lang.get("guild_red_packet_btn_send"))
       -- self._textStageName:setTextColor(Colors.OBVIOUS_YELLOW)
        self._imageRedPacket:loadTexture(Path.getGuildRes("img_lit_hongbao_03"))
    elseif state == GuildConst.GUILD_RED_PACKET_NO_RECEIVE then
       -- self._imageBg:loadTexture(Path.getCommonImage("img_com_board04"))
        self:updateImageView("_imageBg", { visible = true, texture = Path.getCommonImage("img_com_board04") })
        self._textStageName:setString(Lang.get("guild_red_packet_btn_open"))
       -- self._textStageName:setTextColor(Colors.BUTTON_WHITE)
        local bgRes=config.show==1 and "img_lit_hongbao_03" or "img_lit_hongbao_03_2"
        self._imageRedPacket:loadTexture(Path.getGuildRes(bgRes))
    else
        --self._imageBg:loadTexture(Path.getCommonImage("img_com_board04"))
        self:updateImageView("_imageBg", { visible = true, texture = Path.getCommonImage("img_com_board04") })
        self._textStageName:setString(Lang.get("guild_red_packet_btn_see"))
       -- self._textStageName:setTextColor(Colors.BUTTON_WHITE)
        local bgRes=config.show==1 and "img_lit_hongbao_03" or "img_lit_hongbao_04_2"
        self._imageRedPacket:loadTexture(Path.getGuildRes(bgRes))
    end
    local color = config.show == 1 and Colors.OBVIOUS_YELLOW or Colors.CLASS_WHITE
    self._textStageName:setColor(color)
   -- self._resourceNode:setContentSize(cc.size(294,108))
end

function GuildMyRedPacketItemCell:_onButton(sender)
    local offsetX = math.abs(sender:getTouchEndPosition().x - sender:getTouchBeganPosition().x)
	local offsetY = math.abs(sender:getTouchEndPosition().y - sender:getTouchBeganPosition().y)
	if offsetX < 20 and offsetY < 20  then
		self:dispatchCustomCallback()
	end
end

return GuildMyRedPacketItemCell

File: study/ts/script/scene/view/guild/GuildMyRedPacketItemCell.ts
const { ccclass, property } = cc._decorator;

import CommonResourceInfo from '../../../ui/component/CommonResourceInfo'
import ListViewCellBase from '../../../ui/ListViewCellBase';
import { DataConst } from '../../../const/DataConst';
import { Colors } from '../../../init';
import { GuildConst } from '../../../const/GuildConst';
import { Path } from '../../../utils/Path';
import UIHelper from '../../../utils/UIHelper';
import { Lang } from '../../../lang/Lang';
import { TypeConvertHelper } from '../../../utils/TypeConvertHelper';

@ccclass
export default class GuildMyRedPacketItemCell extends ListViewCellBase {

    @property({
        type: cc.Node,
        visible: true
    })
    _resourceNode: cc.Node = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageBg: cc.Sprite = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textRedPacketName: cc.Label = null;

    @property({
        type: CommonResourceInfo,
        visible: true
    })
    _resInfo: CommonResourceInfo = null;

    @property({
        type: cc.Sprite,
        visible: true
    })
    _imageRedPacket: cc.Sprite = null;

    @property({
        type: cc.Label,
        visible: true
    })
    _textStageName: cc.Label = null;

    onCreate() {
        var size = this._resourceNode.getContentSize();
        this.node.setContentSize(size.width, size.height);
        // this._imageRedPacket.setSwallowTouches(false);
    }
    updateData(data) {
        var config = data.getConfig();
        var money = data.getTotal_money() * data.getMultiple();
        var state = data.getRed_bag_state();
        this._textRedPacketName.string = (config.name);
        this._resInfo.updateUI(TypeConvertHelper.TYPE_RESOURCE, DataConst.RES_DIAMOND, money);
        this._resInfo.setTextColor(Colors.BRIGHT_BG_TWO);
        this._resInfo.showResName(false, null);
        if (state == GuildConst.GUILD_RED_PACKET_NO_SEND) {
            UIHelper.loadTexture(this._imageBg, Path.getCommonImage('img_com_board04b'));
            this._textStageName.string = (Lang.get('guild_red_packet_btn_send'));
            UIHelper.loadTexture(this._imageRedPacket, Path.getGuildRes('img_lit_hongbao_03'));
        } else if (state == GuildConst.GUILD_RED_PACKET_NO_RECEIVE) {
            UIHelper.loadTexture(this._imageBg, Path.getCommonImage('img_com_board04'));
            this._textStageName.string = (Lang.get('guild_red_packet_btn_open'));
            var bgRes = config.show == 1 && 'img_lit_hongbao_03' || 'img_lit_hongbao_03_2';
            UIHelper.loadTexture(this._imageRedPacket, Path.getGuildRes(bgRes));
        } else {
            UIHelper.loadTexture(this._imageBg, Path.getCommonImage('img_com_board04'));
            this._textStageName.string = (Lang.get('guild_red_packet_btn_see'));
            var bgRes = config.show == 1 && 'img_lit_hongbao_03' || 'img_lit_hongbao_04_2';
            UIHelper.loadTexture(this._imageRedPacket, Path.getGuildRes(bgRes));
        }
        var color = config.show == 1 && Colors.OBVIOUS_YELLOW || Colors.CLASS_WHITE;
        this._textStageName.node.color = (color);
    }
    onButton(sender) {
        this.dispatchCustomCallback(null);
    }

}